src/bindings/Vscode.ml

```diff
@@ -466,15 +466,20 @@ module LanguageClient = struct
     ; options : processOptions
     }
 
+  type trace =
+    | Off [@bs.as 0]
+    | Messages [@bs.as 1]
+    | Verbose [@bs.as 2]
+  [@@bs.deriving { jsConverter = newType }]
+
   type t =
-    { start : (unit -> unit[@bs])
-    ; stop : (unit -> unit[@bs])
-    ; initializeResult : InitializeResult.t
+    { initializeResult : InitializeResult.t
+    ; mutable trace : abs_trace
```

Comment:
What is the point of saving the `trace` here? As far as I know, the language client is going to pass it automatically to the server anyways. This is the code that I'm referring to:

https://github.com/microsoft/vscode-languageserver-node/blob/master/client/src/common/client.ts#L3249-L3269

Let me know if I'm misunderstanding something.

Reply:
I believe I was the one that misunderstood. I wrote this code under the assumption that [LanguageClient.trace](https://github.com/microsoft/vscode-languageserver-node/blob/master/client/src/common/client.ts#L2772) was the only way to change the trace value. 

I guess I really overthought it :/.
Once I get back to my dev computer, I will remove the redundant code.

Is this behavior documented somewhere? It would be nice to know in the future.

Reply:
The documentation is lacking unfortunately. I learned about this from the official LSP sample https://github.com/microsoft/vscode-extension-samples/blob/master/lsp-sample/client/package.json